pkg/cloudprovider/types.go

```diff
@@ -88,10 +89,18 @@ type InstanceType struct {
 	// Overhead is the amount of resource overhead expected to be used by kubelet and any other system daemons outside
 	// of Kubernetes.
 	Overhead *InstanceTypeOverhead
+
+	once        sync.Once
+	allocatable v1.ResourceList
+}
+
+func (i *InstanceType) precompute() {
```

Comment:
Can we include some detail on why we are precomputing so someone doesn't come along later and think this code can be removed?

Reply:
Done